src/StoreApi/Routes/AbstractRoute.php

```diff
@@ -70,17 +70,69 @@ public function get_response( \WP_REST_Request $request ) {
 					$response = $this->get_route_response( $request );
 					break;
 			}
-			if ( 'GET' !== $request->get_method() && ! is_wp_error( $response ) ) {
-				$response->header( 'X-WC-Store-API-Nonce', wp_create_nonce( 'wc_store_api' ) );
-			}
 		} catch ( RouteException $error ) {
 			$response = $this->get_route_error_response( $error->getErrorCode(), $error->getMessage(), $error->getCode(), $error->getAdditionalData() );
 		} catch ( \Exception $error ) {
 			$response = $this->get_route_error_response( 'unknown_server_error', $error->getMessage(), 500 );
 		}
+
+		if ( is_wp_error( $response ) ) {
+			$response = $this->error_to_response( $response );
+		}
```

Comment:
`rest_post_dispatch` potentially runs on every REST response from all APIs. Converting it ourselves gives us full, contained, control over error responses in the Store API without affecting non-store API endpoints.

Also pointing out that feedback above states:

> Are we sure we want to return a nonce header for every error response?

We would need a way to identify store API requests with this filter, and then further if we wanted to only affect *some* Store API requests. Just pointing this out because it's a downside to using the filter.

Reply:
> An advantage of using the filter is that if we add it to a late priority (eg 999) we also ensure the header is added at the last possible moment?

This is not an advantage against inline header addition which is only done on Store API responses—late addition is of no concern with the current approach.

Another advantage of inline is that our `get_response` methods only ever return valid `WP_REST_Response` objects, rather than __maybe__ returning a `WP_Error` object, so it's more predictable. The code introduced is fairly small and does enable us to further customise the error responses if needed.